### PR TITLE
Support PineLiker in Redis.

### PR DESCRIPTION
**Pull request by wp19991**
change README.md

### CODE REVIEW

**Review by jovany-wang — APPROVED** [review]
Nice work!

File: api/src/main/java/org/pine/api/Pine.java [hunk 1]
```diff
@@ -11,7 +11,9 @@ public static void init(String mode, String address) {
 
     if (mode.equals("distkv")) {
       factoryName = "org.pine.distkvimpl.DistkvImplPineRuntimeFactory";
-    } else {
+    } else if(mode.equals("redis")){
```
Comment (jovany-wang) [inline api/src/main/java/org/pine/api/Pine.java]: Whitespaces between ){

File: api/src/main/java/org/pine/api/Pine.java [hunk 2]
```diff
@@ -28,6 +30,25 @@ public static void init(String mode, String address) {
 
   }
 
+  @SuppressWarnings("unchecked")
+  public static void init(String mode, String address,int index) {
```
Comment (jovany-wang) [inline api/src/main/java/org/pine/api/Pine.java]: I think we should add a new `init` for this. It's better to move this code to the `init` at line8.

File: api/src/main/java/org/pine/api/PineRuntime.java [hunk 3]
```diff
@@ -7,4 +7,6 @@
   PineLiker newLiker();
 
   void connect(String address);
+
+  void select(int index);
```
Comment (jovany-wang) [inline api/src/main/java/org/pine/api/PineRuntime.java]: What does this method do?

File: distkv-impl/pom.xml [hunk 4]
```diff
@@ -20,7 +20,7 @@
         <dependency>
             <groupId>com.distkv</groupId>
             <artifactId>distkv-client</artifactId>
-            <version>0.1.3</version>
+            <version>0.1.4-SNAPSHOT</version>
```
Comment (jovany-wang) [inline distkv-impl/pom.xml]: We don't need to change this.

File: redis-impl/pom.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        </dependency>
+        <dependency>
+            <groupId>redis.clients</groupId>
+            <artifactId>jedis</artifactId>
+            <version>3.1.0</version>
+        </dependency>
+    </dependencies>
+
+
+</project>
```
Comment (jovany-wang) [inline redis-impl/pom.xml]: Add a new line at the end of this file.

File: redis-impl/src/main/java/org/pine/redisimpl/HexadecimalUtil.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package org.pine.redisimpl;
+
+public class HexadecimalUtil {
```
Comment (jovany-wang) [inline redis-impl/src/main/java/org/pine/redisimpl/HexadecimalUtil.java]: I think this method is already in the project.

File: redis-impl/src/main/java/org/pine/redisimpl/PineHandleId.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
+package org.pine.redisimpl;
+
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Random;
+
+public class PineHandleId implements Serializable {
```
Comment (jovany-wang) [inline redis-impl/src/main/java/org/pine/redisimpl/PineHandleId.java]: This method also in this project.

File: redis-impl/src/main/java/org/pine/redisimpl/RedisImplPineRuntimeFactory.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+import org.pine.api.PineRuntime;
+import org.pine.api.PineRuntimeFactory;
+
+public class RedisImplPineRuntimeFactory implements PineRuntimeFactory {
+
+    public PineRuntime createPineRuntime() {
+        return new PineRuntimeRedisImpl();
+    }
+}
```
Comment (jovany-wang) [inline redis-impl/src/main/java/org/pine/redisimpl/RedisImplPineRuntimeFactory.java]: Add a new line at the end of this file.

**Review by jovany-wang — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/pine/api/PineRuntime.java (hunk 3 above)
Comment (wp19991) [reply api/src/main/java/org/pine/api/PineRuntime.java]: Users can select the database stored in redis.
So I added this in the abstract method

**Pull request closed without merging**